### ISSUE TITLE
New horizontal tab design seems very large

### ISSUE THREAD

**Issue by intrnl** [brave/brave-browser#33332]
## Description 

The new horizontal tab design seems very large when compared to what they are *now*, I'd like to ask what's the plan forward when it comes to the old tab design or if there's any plans of introducing a compact mode for this

## Steps to Reproduce

1. Open Brave
2. Visit about:flags
3. Search for brave-horizontal-tabs-update flag and enable it

## Actual result:

![image](https://github.com/brave/brave-browser/assets/20620901/ae4ac979-222d-428e-8178-6d2281bbf70c)

## Expected result:
![image](https://github.com/brave/brave-browser/assets/20620901/aaa0f8fb-b565-4b63-b50f-f5cf0e17d087)

## Reproduces how often: 

Easily reproduced

## Brave version (brave://version info)

1.60.66 Chromium: 118.0.5993.21 (Official Build) nightly (64-bit) 

## Version/Channel Information:

- Can you reproduce this issue with the current release? No.
- Can you reproduce this issue with the beta channel? No.
- Can you reproduce this issue with the nightly channel? Yes.

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? No.
- Does the issue resolve itself when disabling Brave Rewards? No.
- Is the issue reproducible on the latest version of Chrome? No.

## Miscellaneous Information:

Here's how Firefox currently looks
![image](https://github.com/brave/brave-browser/assets/20620901/c4a659a6-ef75-48e2-bd45-c006fa2248a2)

With the compact mode enabled it looks like this
![image](https://github.com/brave/brave-browser/assets/20620901/1d8fb2f3-1bb5-456e-9369-064cc83a04b1)

**Comment by rebron** [comment]
cc: @aguscruiz

**Comment by aeimi** [comment]
Size and style of tabs is one issue. But dead margins around the tabs are huge UX mistake. Which is also visible even in stable vertical tabs  They have dead margins around them and lack of hover background state.

Firefox have big tabs like new tab style in Brave dev/nightly, but all spaces even around tabs and between them up to the end of the screen are active/clickable. 

Tabs either vertical and horizontal MUST be active to hover/click edge to edge with screen and each other. No dead margins/puddings.  Quick and intuitive tab switching with mouse depend highly in my case on edge of screen.  Why you broke most fundamental thing on my browser use workflow I don't know.  That's why I for a first time post on Brave issues because this is so important for me to be sure I always will be able to turn off new design and use old one. Even with a flag. But I was afraid you may some day take away this option for me and force me to use this broken tabs design. 

Please don't take away something that is perfect. Current horizontal tab design is what I'm 100% happy with.  New design should be optional in settings. Or at least by flag.  Still vertical tabs should be fixed as I mentioned above. 

Horizontal space on laptops and huge amount of desktops are limited. There is no real reason to add padding to tabs when font size is the same. Some may like it so ok, but for me it's just waste of display space, nothing more.

**Comment by bradleyrichter** [comment]
@aeimi  Thanks for the detailed description. It appears that we have 3 pixels of dead space between horizontal and vertical tabs. We will look into it.

**Comment by zenparsing** [comment]
We've created an issue to track the "dead space between tabs" for horizontal tabs here: https://github.com/brave/brave-browser/issues/34190.

**Comment by aeimi** [comment]
@bradleyrichter Thank you. Just to be clear. While in full size window (Windows OS) Dead margin **between tabs and edge of the screen is worst**.  In vertical tabs lack of hover background should also be fixed as it is in horizontal tabs. 

I was worried. Because many of users just would not tell you this. All Chromium browsers and even Firefox by default have tabs to the edge of screen. Especially new users trying Brave might be turn off because tabs is ~50% of visible UI and crucial for mouse navigation. 

For me myself it is crucial issue. I'm using Brave as fast I heard it switched to Chromium.  And many of it's added features doesn't bother me. But take away something that it right without option to switch back, will be deal breaker for me.

I get, I just one guy. But think about mass users. They also have strong already workflow and they don't think about it until something doesn't work the way they already know it. Chrome, Edge, Opera, Vivaldi, etc etc.  Big tabs with big padding are just a waste of space on smaller screens. Especially laptops.  Font size is the same so no improvement readability. So why just add empty spaces? Form should always follow function first.

Some 16:9 screens especially on laptops as I said are really struggle in matter of vertical space. Give users at least an option. This should be hard. Don't think about just one me who is reporting about this. But about global share market of your users. People just walk away and in most cases will not tell you why. We all use things mindlessly. When some things make us think there is a problem. Things must be intuitive/natural.

**Comment by bradleyrichter** [comment]
@aeimi  Totally agree with most of your points. We have a PR in the works to fix the top margin gap in windows expanded and full screen views. I agree that is really important for usability as the mouse naturally stops at the top of the monitor. If I recall we had fixed this a few years back after a similar UI update created the same problem.

@zenparsing - Could you please link it here?

Regarding space from loosening up the UI, many UI elements are trending larger to both appear easier to use and allow for larger hit targets. We get constant requests to make URL bars taller on all platforms.  UI elements for grouping have necessitated adding a bit of additional space as well.

Believe it or not, before we switched to Chromium I had spent a bunch of time hand-tuning the elements in the toolbars/tabs in order to make them as short as possible. Brave was the shortest for a while. Since then we have all enjoyed getting more pixels and higher screen densities so naturally things needed to loosen up.

This time we we were careful to be aligned with Chrome, Firefox and Opera for stack height. We can certainly investigate adding a compact mode as you requested for more advanced users.

**Comment by aeimi** [comment]
@bradleyrichter Yeah that's what I guessed. You switched to High DPI. But many of users are still on standard. That's why it should be optional (even if not default). Especially you did good job as you describe to finetune it for standard monitors. Desktop monitors with 4k are still in premium category.  It could be defaulted to new tab style on High DPI and keep current style on standard low DPI monitors. But option should remain in settings anyway for users to decide. 

I just found this https://www.reddit.com/r/chrome/comments/17ql8mz/time_to_say_goodbye_to_chrome/ while I was trying to figure out about my other issue in Brave with double clicking. ... hmm. But that's another issue. Not related to this one. 

Good example what I wrote about in previous post here. Users aren't happy when you take away from them what they like without any option to get it back. Especially, when it should not be big technical issue to implement such setting/option.

I like Brave very much and I don't want ever to looking around for any alternative. 

> Regarding space from loosening up the UI, many UI elements are trending larger to both appear easier to use and allow for larger hit targets. We get constant requests to make URL bars taller on all platforms. UI elements for grouping have necessitated adding a bit of additional space as well.

Every time when I heard you implement something that some users like I like it very much. But I was worried that no one cares anymore in Brave and as many good projects that doesn't care about it's users Brave will die with time. And that would be sad for me. Since I'm a fan of Brave for more than one reason. More like, heart invested in it :) History, reasons, logo, name.

**Comment by intrnl** [comment]
> Since then we have all enjoyed getting more pixels and higher screen densities so naturally things needed to loosen up.
"All" seems pretty loose here, bearing in mind that 1366x768 or the equivalent 1920x1080 at 125% is **very** common. I'd really prefer the option of keeping both.

**Comment by aeimi** [comment]
Default setup should be for most people. But settings are for willing. Customization and flexibility should be always paradigm in building software for masses.  i.e. Even there is no officially released tab scrolling, but I always use this enabled by Flag.

Vertical tabs have those unwanted margin/gaps (between them and from edge of screen) also and lack of proper hover state as I mentioned before.

**Comment by aeimi** [comment]
New nightly fixes top edge support for tabs :) 🎉 That was #1 feature on my list to fix. Thanks.

> Brave nightly 1.62.38 Chromium: 119.0.6045.134

**Comment by intrnl** [comment]
seems like there's borders on the page content now :/

![image](https://github.com/brave/brave-browser/assets/20620901/cca1f5fe-a1af-4565-8890-ecdc11b7d794)

**Comment by aeimi** [comment]
@bradleyrichter in dev 1.62.93 we lost original colors. background grays and orange borders and some sizes. All that, still is, in my opinion, perfect in current stable. But some of that I can restore by building custom skin. Yet original size of address bar field. is bigger now. After disabling _Updated horizontal tabs design_ flag. Because new tabs still have inactive gap between them while you move on the edge. I ping to you because you said you crafted original with attention. so who can I call for that. I will probably want this original design even when I will switch to 4k for some time. Even if we have some option to manually set some spacing/padding/font sizes even in skin or in settings. But.. whatever. I would be happy if there was any way for user to keep or restore manually current stable design.

**Comment by intrnl** [comment]
> seems like there's borders on the page content now :/

I've made a separate issue for this one because it makes reaching for the scrollbar harder as I can't just slam my mouse cursor to the edge of the screen anymore. It's practically not a good design if it ruins mouse interactions like this. https://github.com/brave/brave-browser/issues/34679

**Comment by aeimi** [comment]
iBorders - pure nonsense

**Comment by fhoshino** [comment]
Backing this suggestion up, really don't want to have a fat tab bar especially I'm not on a touchscreen.
Those borders are really unnecessary. At least give us an option in the setting page to disable the new tab look.

**Comment by MrVoltz** [comment]
I'd also want an option to make the tabs and address/toolbar smaller vertically. The address bar itself has unnecessarily large padding, it is precious vertical real estate. Just give us a toggle in settings to make this stuff smaller.

**Comment by MrVoltz** [comment]
![Screenshot from 2024-02-07 20-30-09](https://github.com/brave/brave-browser/assets/5212649/03917cd9-952e-4dca-aa32-2131f7f03e11)
Brave on the left, Firefox in the middle, Chrome on the right. Both Brave and Firefox are more wasteful than Chrome.

**Comment by jjspace** [comment]
I want to add that the new design looks even worse with my theme (that I've used for many years) by creating 3 layers of background colors because of the extra padding. Why do we need that extra space to create a bubble for every tab?

 
![2024-02-07_17-44-04](https://github.com/brave/brave-browser/assets/8007967/6b940023-6e5a-4bb8-af2b-327d239480f2)

**Comment by will-stone** [comment]
@MrVoltz Mind posting an updated screenshot with the inclusion of Brave's old style? It'll help highlight the original issue that @intrnl was raising.

**Comment by zenparsing** [comment]
@jjspace We've created a new issue to track the theming problem: https://github.com/brave/brave-browser/issues/35905

**Comment by aeimi** [comment]
Props for removing dead unclickable margins. Thanks!

I hope you don't plan to remove classic tabs view from flags?

**Comment by aeimi** [comment]
Classic tabs.  In Brave  1.69.62
With disabled  flag: _Updated horizontal tabs design_
Now old design is messy to the level it's unusable not just because of weird sizing but reintroduced dead margins.

I'm still asking for not dropping old design. Even if it's just under flag. Although I would still encourage to add option in official settings to choose between classic tabs and new updated version. Or at least option to get rid of unneeded margins/padding. Like option to select between Standard and Compact mode. It's not about tabs or pill style. But more about space they use.  Is there any chance? @bradleyrichter

**Comment by aeimi** [comment]
And here it is. Classic tabs broken already in beta release version 1.68.115. Will anybody care about this @rebron ?

**Comment by MrVoltz** [comment]
Could you please post a screenshot of the broken state? Everything is still ok in stable.

**Comment by aeimi** [comment]
**In stable it's currently still OK**. But in _beta_ as I wrote above, no longer. So I was afraid it will be moved to brake stable next.
![image](https://github.com/user-attachments/assets/0da82a7a-2511-4f2d-ad2e-dfe307440fb8)

**Comment by aeimi** [comment]
> Everything is still ok in stable.

Not anymore. Here after latest update to chromium 127, latest STABLE od Brave update bring in broken classic tabs with 1.68.128 

That's what I was afraid of. I don't know if anybody care about this anymore in Brave team or not.

**Comment by aguscruiz** [comment]
Hi all, we've been looking at the feedback given here, thanks! I wanted to address a couple updates we have coming:

### Colors and themes
Regarding the colors and themes, we're moving our theming engine to match what's coming upstream from Chromium. We're working hard to adapt our current themes to that new engine, and it's a work in progress, so some things may not look ok until we can polish everything. 

We're sorry for that, but it's gonna allow us to create a more consistent theme and fix our color themes to make them much nicer.

### New tab shapes
We're moving away from the classic tab shape in favor of the floating ones due to new features that we're developing, like split view and others, that just don't work with the classic shape. 

Issues regarding dead spaces and what are clearly bugs will be address until they work perfectly fine and how they're are expected. But at some point, the flag for keeping the classic tab shape will cause the UI to break if you use some of the new features we have in the pipeline. 

It's not that we don't care about the classic design, but it's something we need to evolve from in order to evolve the browser features. 

### Content window borders
Just like the case of the new tab shape, that border around the window is something we'll need in order to create a better experience for our new features. 


I hope that clarifies some of the issues. We're actively looking at your feedback here and trying to minimize the changes and address your pain points though. Just know that we do care 🧡

**Comment by 097115** [comment]
> I hope that clarifies some of the issues

Yeah, *pretty much*. I guess, I'll just stick with 1.66.118 for as much as possible.

**Comment by aeimi** [comment]
Thanks for answer @aguscruiz  
I understand that new features may require new approach. Just remember, maybe you'll find a way to have officially supported in settings for a "compact" mode tabs style. I already switched to new tab style after latest 127 stable update, since they are more usable than broken classic tabs. If classic tabs are already dead end, the only visual difference I ask for is less white space padding around text on tab and less margin around them. Zero visual margin classic tabs had ;)  That's all. They are already usable since you fixed unclickable zones.

**Comment by aeimi** [comment]
> Yeah, _pretty much_. I guess, I'll just stick with 1.66.118 for as much as possible.

No future.

**Comment by nastavoi** [comment]
Here's my two cents - I honestly don't mind the new tab shape, I've adjusted pretty quickly but can't wait for the compact mode since I feel they are indeed too big.

As for the window border - please consider making them clickable / usable with vertical scrollbar so that the user would be able to click on it by moving the mouse cursor to the edge of the screen, without looking at it. The current implementation of **#brave-web-view-rounded-corners** disrupts my (and I guess many people) workflow as I own UWQHD monitor (#34496).

**Comment by aeimi** [comment]
Active border/hoverable sidebar have nothing to do with additional border/shadow feature. Borders and shadow of window should be system wide feature. Not forced style of one program.

**Comment by aeimi** [comment]
Compact mode is here! Thank you! Even classic tabs are fixed. Nice!
Although I think, because of visual balance I might even prefer new style. 

It's really minor, visual thing, but If you plan to continue with classic tabs style, I think top corner radius should be reduced to minimal like before.  But that's just a detail. There is no dead zones and that's what most important. Thanks!

**Comment by fhoshino** [comment]
Really enjoying the compact mode, now claiming back a few more pixel of web content for each scroll.

**Comment by aeimi** [comment]
**Compact mode for the browser - Prototype**
This feature is a prototype. Reduces the height of the browser controls (tabstrip, toolbar, bookmarks bar, etc.) to allow more space for the web contents  
[#compact-mode](brave://flags/#compact-mode)

This flag seems to not be make any changes yet I can notice. But I'm glad to see it's there. I guess. WIP.

**Comment by aeimi** [comment]
Compact mode for the browser - Prototype
This feature is a prototype. Reduces the height of the browser controls (tabstrip, toolbar, bookmarks bar, etc.) to allow more space for the web contents
#compact-mode

This flag seems to not be make any changes yet I can notice. But I'm glad to see it's there. I guess. WIP.
It would be great to see second row compacted like in old design.

**Comment by aeimi** [comment]
**Compact mode for the browser - Prototype**
This feature is a prototype. Reduces the height of the browser controls (tabstrip, toolbar, bookmarks bar, etc.) to allow more space for the web contents

This flag seems to not be make any changes yet I can notice. But I'm glad to see it's there. I guess. WIP.
It would be great to see second row compacted too, like in old design.

**Comment by bsclifton** [comment]
Going to close this as stale as there hasn't been any recent updates

Please leave a comment or open a new issue if there are still concerns. Thanks!
